### DISCUSSION TITLE
Where is the scheduler code in the source code for the latest version of K3S？

### DISCUSSION THREAD

**Discussion by Jiaxuan-C** [k3s-io/k3s#4999] [Q&A]
Where is the scheduler code in the source code for the latest version of K3S？Does the latest version of the K3S scheduler support the same scheduling framework as the K8S？

I really want to know, this might be an easy question for you, but for a beginner，emm，please help the child................

**Comment by brandond** [comment]
K3s is a Kubernetes distribution, which means we just build and package the code from upstream. The Kubernetes scheduler code lives at https://github.com/kubernetes/kubernetes/tree/master/cmd/kube-scheduler

**Reply by Jiaxuan-C** [reply]
thanks for your reply. But if I want to customize a new scheduling policy for K3S, does that mean I can use the K8S scheduling framework, implement it as a plugin, and apply it directly to K3S?

**Reply by brandond** [reply]
You could probably build a version of K3s that used your scheduler instead of the default one, yes. That's not something I would expect to tackle as a beginner, though.

**Reply by Jiaxuan-C** [reply]
haha, thanks very much. This is related to my graduation project , so I will continue to learn it. thanks for your patience
